lib/src/theme_provider_suggestors/theme_provider_adder.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:codemod/codemod.dart';
+import 'package:collection/collection.dart' show IterableExtension;
+import 'package:over_react_codemod/src/util/component_usage.dart';
+
+/// Suggestor that wraps the contents of `react_dom.render` calls in a ThemeProvider.
+class ThemeProviderAdder extends GeneralizingAstVisitor
+    with AstVisitingSuggestor {
+  /// The theme that will be added to the ThemeProvider.
+  final String theme;
+
+  ThemeProviderAdder(this.theme);
+
+  @override
+  visitMethodInvocation(MethodInvocation node) {
+    super.visitMethodInvocation(node);
+
+    // Don't update if [node] is not a render call or the file is in a test directory.
+    if (node.methodName.name != 'render' ||
+        context.relativePath.split('/').contains('test')) return;
```

Comment:
Note that this check could potentially skip over [subdirectories named `test` not in the top-level `test` directory](https://sourcegraph.wk-dev.wdesk.org/search?q=file:lib%5Cb.*/test/+lang:dart+-repo:cp_labs+-repo:messaging-sdk+-repo:puppeteer-platform+-repo:frugal&patternType=literal).

From the linked usages, this shouldn't be an issue, but I think it'd be worth calling out in a code comment